water_neuron/src/state_machine/mod.rs

```diff
@@ -1407,6 +1426,376 @@ fn e2e_basic() {
     assert_eq!(full_neuron.cached_neuron_stake_e8s, 1000155632);
 }
 
+#[test]
+fn should_cancel_withdrawal() {
```

Comment:
Nit: it is better to have multiple independent/isolated tests each checking their own scenario rather that one test that combines multiple scenarios.